addons/binding/org.openhab.binding.jeelink/src/main/java/org/openhab/binding/jeelink/handler/JeeLinkSensorHandler.java

```diff
@@ -0,0 +1,221 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.jeelink.handler;
+
+import static org.openhab.binding.jeelink.JeeLinkBindingConstants.*;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.builder.ThingBuilder;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.jeelink.config.JeeLinkSensorConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Abstract thing handler for sensors connected to a JeeLink.
+ *
+ * @author Volker Bier - Initial contribution
+ */
+public abstract class JeeLinkSensorHandler<R extends Reading> extends BaseThingHandler {
+    private Logger logger = LoggerFactory.getLogger(JeeLinkSensorHandler.class);
+
+    private JeeLinkReadingListener<R> listener;
+    private Class<R> clazz;
+
+    private ScheduledFuture<?> valueUpdateJob;
+    private ScheduledFuture<?> statusUpdateJob;
+
+    private boolean initialized;
+
+    public JeeLinkSensorHandler(Thing thing, Class<R> clazz) {
+        super(thing);
+
+        this.clazz = clazz;
+    }
+
+    @Override
+    public synchronized void handleCommand(ChannelUID channelUid, Command command) {
+        if (command instanceof StringType && channelUid.getIdWithoutGroup().equals(SENSOR_ID_CHANNEL)) {
+            final String cmdName = command.toString();
+
+            Matcher matcher = Pattern.compile("SetSensorId (\\w+)").matcher(cmdName);
+            if (matcher.matches()) {
+                final String oldId = getThing().getProperties().get(PROPERTY_SENSOR_ID);
```

Comment:
What is going on here? Channels definitely must not be used to set Thing properties... What's the use case here?